python/lsst/galaxy_shear/processShearTest.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 20082014 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import numpy
+import lsst.daf.base
+import lsst.pex.config
+import lsst.pipe.base
+import lsst.afw.table
+import lsst.afw.image
+import lsst.afw.math
+import lsst.meas.algorithms
+
+from lsst.obs.great3.processBase import *
+
+class ProcessShearTestConfig(ProcessBaseConfig):
+    pass
+"""
+   This subclass obs.great3.ProcessBaseTask is used to call a measurement algorithm,
+   assumed to be the only algorithm which is not a Centroid algorithm, using as its
+   sources a GalSim catalog provided by the "epoch_catalog" for the dataId
+"""
+class ProcessShearTestTask(ProcessBaseTask):
+
+    ConfigClass = ProcessShearTestConfig
+
+    _DefaultName = "processShearTest"
+
+    def __init__(self, **kwds):
+        ProcessBaseTask.__init__(self, **kwds)
+        self.schema.addField("psf_index", type = int, doc = "index of psf within library")
+        self.schema.addField("psf_library", type = int, doc = "number of psf_libary_nn.fits")
+        self.centroidPlugin = None
+        self.measPlugin = None
+        for plugin in self.measurement.plugins.keys():
+            if plugin.find("Centroid") > 0:
+                self.centroidPlugin = self.measurement.plugins[plugin]
+            else:
+                self.measPlugin = self.measurement.plugins[plugin]
+
+    #   Transfer information already know about the sources from the GalSim epoch catalog
+    #   to an afwTable catalog.  The Psf library and index in particular are needed.
+    def buildSourceCatalog(self, imageBBox, dataRef):
+        """Build an empty source catalog, using the provided sim catalog's position to generate
+        square Footprints and its ID to set that of the source catalog.
+        """
+        sourceCat = lsst.afw.table.SourceCatalog(self.schema)
+        sourceCat.getTable().setMetadata(self.algMetadata)
+        simCat = dataRef.get("epoch_catalog", immediate=True)
+        xKey = simCat.schema.find('x').key
+        yKey = simCat.schema.find('y').key
+        idKey = simCat.schema.find('ID').key
+        nGals = imageBBox.getWidth() / self.config.galaxyStampSize
```

Comment:
Probably best to use `//` here with `from __future__ import division`, so this'll be easy to port to Python 3 in the future.